lib/teleagent/agent.go

```diff
@@ -14,16 +14,63 @@ import (
 	"golang.org/x/crypto/ssh/agent"
 )
 
+// Agent extends the agent.Agent interface.
+// APIs which accept this interface promise to
+// call `Close()` when they are done using the
+// supplied agent.
+type Agent interface {
+	agent.Agent
+	io.Closer
+}
+
+// wrapper wraps an agent.Agent in the extended
+// Agent interface.
+type wrapper struct {
+	agent.Agent
+}
+
+func (w wrapper) Close() error { return nil }
+
+// WrapAgent wraps an instance of the standard agent.Agent interface
+// in the extended Agent interface.  Note that calling Close on the
+// resulting Agent is a NOP, even if the underlying type of the
+// supplied agent.Agent instance has a Close method.  This means that
+// WrapAgent can also be called on an instance of the extended Agent
+// interface in order to protect its Close method from being called.
+func WrapAgent(std agent.Agent) Agent {
+	return wrapper{std}
+}
+
+// AgentGetter is a function used to get an agent instance.
+type AgentGetter func() (Agent, error)
```

Comment:
how about just `Getter` to make this `teleagent.Getter`?
https://github.com/golang/go/wiki/CodeReviewComments#package-names